browser/src/core/Socket.js

```diff
@@ -1476,6 +1476,9 @@ app.definitions.Socket = L.Class.extend({
 
 	_onSocketClose: function () {
 		console.debug('_onSocketClose:');
+		if (this.ReconnectCount > 0)
+			return;
```

Comment:
I worry that we still need some of the other flags below to be set for some reconnection logic to work correct.

For example, if we don't removeAllViews() here, and we do reconnect, what/who will removeAllViews() later? Same question for resetting other objects.

I fear that the logic always assumes that reconnection will have a reset state, but here we are skipping resetting.

How was this tested? Is there a clear scenario that needs this fix? Could this cause other regressions that we haven't thought about yet?

Reply:
Interesting questions:

				// We're connected: cancel timer and dialog.
				this.ReconnectCount = 0;

do you mean Reset the counter?

did you find a corner case removeAllViews()?
can you provide me logs or data or steps to reproduce as evidence?

steps to test:

1) load the document
2) force an interrupt the service (whatever method you like)
3) the re-connection starts..

Reply:
@hcvcastro, sorry, by "reset" I mean to set the values of the state members to a known value.

For example, `this._map._active = false;`. If we skip this, the code that depends on `map._active` will see it as `true` and will assume we are connected/active.

Thanks for the testing steps. Here are some more specific scenarios that we should make sure work correctly.

What happens when reconnection doesn't work (but the document was loaded)? (This will happen when the server crashes.)
What happens when loading the document doesn't work, and we have to retry? (Do we give up now? Do we give up after retrying a few times?)
What happens when the server is shut down (cleanly) after loading a doc, and then started up, after a minute or two? (This is a supported "recycling" scenario.)

If the above scenarios work correctly, then we should be good. Please confirm they work after trying multiple times, to be sure.

Thanks.